src/Native/Unix/System.Native/pal_time.c

```diff
@@ -152,3 +153,82 @@ int32_t SystemNative_GetTimebaseInfo(uint32_t* numer, uint32_t* denom)
     }
     return 1;
 }
+
+#if defined(_ARM_) || defined(_ARM64_)
+#define SYSCONF_GET_NUMPROCS _SC_NPROCESSORS_CONF
+#else
+#define SYSCONF_GET_NUMPROCS _SC_NPROCESSORS_ONLN
+#endif
+
+int32_t SystemNative_GetCpuUtilization(ProcessCpuInformation* previousCpuInfo)
+{
+    static long numProcessors = 0;
+
+    if (numProcessors <= 0)
+    {
+        numProcessors = sysconf(SYSCONF_GET_NUMPROCS);
+        if (numProcessors <= 0)
+        {
+            return 0;
+        }
+    }
+
+    uint64_t kernelTime = 0;
+    uint64_t userTime = 0;
+
+    struct rusage resUsage;
+    if (getrusage(RUSAGE_SELF, &resUsage) == -1)
+    {
+        assert(false);
+        return 0;
```

Comment:
Should we zero out the values in previousCpuInfo in this case?  Or it's ok to just leave the previous values that were passed in?  Presumably this won't happen, anyway, hence the assert.

Reply:
Should not be necessary. If the function succeeds at any later time the calculations will just work and account for the time during the failed call.

Reply:
Ok.